File: Test/AdventureWorksFunctionalModel/Person/ContactType.cs







using System;
using NakedFunctions;

namespace AW.Types {
    [Bounded]
    public class ContactType : IHasModifiedDate {
        [Hidden]
        public int ContactTypeID { get; init; }

        [MemberOrder(1)]
        public string Name { get; init; } = "";

        
        [MemberOrder(99)]
        [Versioned]
        public DateTime ModifiedDate { get; init; }

        public override string ToString() => Name;


    }
}

File: Test/AdventureWorksLegacyModelVB/Person/ContactType.vb
Namespace AW.Types

	<Bounded>
	Partial Public Class ContactType
		Implements IHasModifiedDate

		<Hidden>
		Public Property ContactTypeID() As Integer

		<MemberOrder(1)>
		Public Property Name() As String = ""

		<MemberOrder(99)>
		Public Property ModifiedDate() As DateTime Implements IHasModifiedDate.ModifiedDate

		Public Overrides Function ToString() As String
			Return Name
		End Function

	End Class
End Namespace